Build with --enable-relocatable on FreeBSD

On FreeBSD (11.0-STABLE, amd64, r313735) build failed with --enable-relocatable:

> make
[skip]
../liblepton/src/.libs/liblepton.so: undefined reference to `canonicalize_file_name'
make[4]: stopped in /home/dmn/lepton/lepton.git/gaf

  Then I found guess_install_data_dir() function in
liblepton/src/edapaths.c and there:
1) canonicalize_file_name() in a non-portable GNU extension
2) there is no /proc/self/exe on FreeBSD (there is /proc/curproc/file
symlink, however). In fact, the system and all programs are functionig
perfectly well even when /proc is unmounted.
  The following discussion on stackoverflow.com may be of help:
http://stackoverflow.com/questions/1023306/finding-current-executables-path-without-proc-self-exe

Thank you for your work.
Regards,
Dmitry.

--- a/NEWS.md
+++ b/NEWS.md
@@ -165,7 +165,7 @@ Please check `info geda-scheme` for full details.
 
 - The configure script now supports an `--enable-relocatable` switch.
   When specified, Lepton tools do not have the configured prefix
-  compiled into them.
+  compiled into them. This option is not supported on BSD systems.
 
 - The configure script now supports a `--disable-deprecated` switch.
   When specified, various deprecated features and behaviour are

--- a/m4/geda-relocatable.m4
+++ b/m4/geda-relocatable.m4
@@ -24,7 +24,7 @@ AC_DEFUN([AX_OPTION_RELOCATABLE],dnl
 
   AC_MSG_CHECKING([whether to do a relocatable build])
   AC_ARG_ENABLE([relocatable],
-    [AS_HELP_STRING([--enable-relocatable], [build relocatable binaries])],
+    [AS_HELP_STRING([--enable-relocatable], [build relocatable binaries (not supported on BSD systems)])],
     [], [enable_relocatable="no"])
 
   if test "X$enable_relocatable" = "Xyes"; then

